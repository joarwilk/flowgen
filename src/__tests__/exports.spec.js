// @flow

import { compiler, beautify } from "..";

it("should handle exports", () => {
  const ts = `
export default module
export { module }
export { module as newModule }
export { GeneratorOptions } from "@babel/generator";
export { GeneratorOptions as NewGeneratorOptions } from "@babel/generator";
export * from 'typescript';
//enable when typescript supports
//export traverse, { Visitor, NodePath } from "@babel/traverse";
//export template from "@babel/template";
//export * as t from "@babel/types";`;
  const result = compiler.compileDefinitionString(ts, { quiet: true });
  expect(beautify(result)).toMatchSnapshot();
});

test("should handle unnamed default export", () => {
  const ts = `
export default function(): void;
`;
  const result = compiler.compileDefinitionString(ts, { quiet: true });
  expect(beautify(result)).toMatchSnapshot();
});

test("should handle export declare const", () => {
  const ts = `
export declare const foo: {
  foo: {
    bar: 'thing',
    baz: 'thing',
  }
};
  `;
  const result = compiler.compileDefinitionString(ts, { quiet: true });
  expect(beautify(result)).toMatchSnapshot();
});
