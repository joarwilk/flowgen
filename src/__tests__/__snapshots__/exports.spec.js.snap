// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle export declare const 1`] = `
"declare export var foo: {
  foo: {
    bar: \\"thing\\",
    baz: \\"thing\\",
    ...
  },
  ...
};
"
`;

exports[`should handle exports 1`] = `
"declare export default typeof module;
declare export { module };
declare export { module as newModule };
declare export { GeneratorOptions } from \\"@babel/generator\\";
declare export {
  GeneratorOptions as NewGeneratorOptions
} from \\"@babel/generator\\";
declare export * from \\"typescript\\"
"
`;

exports[`should handle unnamed default export 1`] = `
"declare export default function fn(): void;
"
`;
